fix(css): update 'drop-shadow()' page (#29013)
mdn/content: files/en-us/web/css/box-shadow/index.md

Author: done

### Values

- `<color>` {{optional_inline}}

  - : Specifies color for the shadow. See {{cssxref("&lt;color&gt;")}} values for possible keywords and notations.
    If not specified, the value of the {{cssxref("color")}} property defined in the parent element is used.

- `<length>`

  - : Specifies the offset length of the shadow. The four values are interpreted as follows:

>     - The first {{cssxref("&lt;length&gt;")}} value is interpreted as `<offset-x>` (horizontal offset). Negative values place the shadow to the left of the element. If not specified, the value of `0` is used for the missing length.
>     - The second {{cssxref("&lt;length&gt;")}} values is interpreted as `<offset-y>` (vertical offset) values. Negative values place the shadow above the element. If not specified, the value of `0` is used for the missing length.\
>       If both `<offset-x>` and `<offset-y>` are set to `0`, the shadow is placed behind the element (and may generate a blur effect if `<blur-radius>` and/or `<spread-radius>` is set).
>     - The third optional {{cssxref("&lt;length&gt;")}} values is interpreted as `<blur-radius>`. The larger this value, the bigger the blur, so the shadow becomes bigger and lighter. Negative values are not allowed. If not specified, it will be `0` (the shadow's edge is sharp). The specification does not include an exact algorithm for how the blur radius should be calculated, however, it does elaborate as follows:

Reviewer: Suggesting the same style as on the `drop-shadow()` page

```suggestion
    - If two values are specified, they are interpreted as `<offset-x>` (horizontal offset) and `<offset-y>` (vertical offset) values. Negative `<offset-x>` value places the shadow to the left of the element. Negative `<offset-y>` value places the shadow above the element.\
      If not specified, the value of `0` is used for the missing length. If both `<offset-x>` and `<offset-y>` are set to `0`, the shadow is placed behind the element (and may generate a blur effect if `<blur-radius>` and/or `<spread-radius>` is set).
    - If three values are specified, the third value is interpreted as `<blur-radius>`. The larger this value, the bigger the blur, so the shadow becomes bigger and lighter. Negative values are not allowed. If not specified, it will be set to`0` (meaning that the shadow's edge will be sharp). The specification does not include an exact algorithm for how the blur radius should be calculated; however, it does elaborate as follows:
```

      > …for a long, straight shadow edge, this should create a color transition the length of the blur distance that is perpendicular to and centered on the shadow's edge, and that ranges from the full shadow color at the radius endpoint inside the shadow to fully transparent at the endpoint outside it.

>     - The fourth optional {{cssxref("&lt;length&gt;")}} value is interpreted as `<spread-radius>`. Positive values will cause the shadow to expand and grow bigger, negative values will cause the shadow to shrink. If not specified, it will be `0` (the shadow will be the same size as the element).

Reviewer: ```suggestion
    - If four values are specified, the fourth value is interpreted as `<spread-radius>`. Positive values will cause the shadow to expand and grow bigger, negative values will cause the shadow to shrink. If not specified, it will be set to `0` (that is, the shadow will be the same size as the element).
```

Reviewer: While we're updating the value descriptions, can we also update this part just before "Values" to remove duplicate info:

> Specify a single box-shadow using:
>  - Two, three, or four `<length>` values.
>     - If only two values are given, they are interpreted as `<offset-x>` and `<offset-y>` values.
>     -  If a third value is given, it is interpreted as a `<blur-radius>`.
>     - If a fourth value is given, it is interpreted as a `<spread-radius>`.
>  - Optionally, the `inset` keyword.
>   - Optionally, a `<color>` value.
> 
> To specify multiple shadows, provide a comma-separated list of shadows.

- We should probably move the above text to the "Values" section.
- And update it to:
> You can create a single box-shadow by specifying two, three, or four `<length>` values, an optional `inset`, and an optional `<color>` value. To create multiple box-shadows, specify a comma-separated list of shadows for the `box-shadow` property.

    > …for a long, straight shadow edge, this should create a color transition the length of the blur distance that is perpendicular to and centered on the shadow's edge, and that ranges from the full shadow color at the radius endpoint inside the shadow to fully transparent at the endpoint outside it.

- `<spread-radius>` {{optional_inline}}

  - : This is a fourth {{cssxref("&lt;length&gt;")}} value. Positive values will cause the shadow to expand and grow bigger, negative values will cause the shadow to shrink. If not specified, it will be `0` (the shadow will be the same size as the element).

- `<color>` {{optional_inline}}

  - : See {{cssxref("&lt;color&gt;")}} values for possible keywords and notations.
    If not specified, it defaults to {{cssxref("&lt;color&gt;","currentcolor","#currentcolor_keyword")}}.

- `inset` {{optional_inline}}

>   - : If not specified (default), the shadow is assumed to be a drop shadow (as if the box were raised above the content).
>     The presence of the `inset` keyword changes the shadow to one inside the frame (as if the content was debossed inside the box). Inset shadows are drawn inside the border (even transparent ones), above the background, but below content.

Reviewer: Starting the description to explain what it does or is used for

```suggestion
  - : Changes the shadow from an outer box-shadow to an inner box-shadow (as if the content is pressed into the box). Inset shadows are drawn inside the box's border (even if the border is transparent), and they appear above the background but below the content. By default, the shadow behaves like a drop shadow, giving the appearance that the box is elevated above its content. This is the default behavior when `inset` is not specified.
```

blockquote {
  padding: 20px;
  box-shadow:
    inset 0 -3em 3em rgba(0, 200, 0, 0.3),
    0 0 0 2px white,

>     0.3em 0.3em 1em rgba(200, 0, 0, 0.6);

Reviewer: This is looking great. Thanks!
